[Bug]: new warning for package source mappings doesn't pass a value for the resource string placeholder

### NuGet Product Used

MSBuild.exe

### Product Version

Microsoft (R) Build Engine version 17.2.0-preview-22178-03+1160f4e0e for .NET Framework

### Worked before?

_No response_

### Impact

It bothers me. A fix would be nice

### Repro Steps & Context

Updated to a VS build containing #11505.  Someone on my team started seeing this error (because we have Warnings As Errors, it broke the build):

> error NU1507: There are {0} package sources defined in your configuration. When using central package management, please map your package sources with package source mapping (https://aka.ms/nuget-package-source-mapping) or specify a single package source. [d:\dd\WebTools\WebTools.sln]

Note: the string still has {0} because it doesn't pass a value to the placeholder.  [Source](https://github.com/NuGet/NuGet.Client/blob/b30bcfb30ffa5d6fad591737fe6055b6ff6d01b7/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L476)

Additional comments:
* Our nuget.config only has a single source.  We're perplexed about why we're even getting this warning.  Maybe some additional info about which feeds are being detected would also be helpful?  NuGet.config contents:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <clear />
    <add key="WebTools VSTS Feed" value="https://devdiv.pkgs.visualstudio.com/...snip..." />
  </packageSources>

  <!-- Don't let user disabled sources in VS prevent us from restoring -->
  <disabledPackageSources>
    <clear />
  </disabledPackageSources>
</configuration>
```
* This isn't happening consistently.  I couldn't repro on my machine, same VS build, same repo. 🤷‍♂️ There may be something different in our environments, but the message isn't helpful for us to figure out what.

### Verbose Logs

_No response_

--- a/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs
+++ b/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs
@@ -473,7 +473,7 @@ namespace NuGet.Commands
             // Log a warning if there are more than one configured source and package source mapping is not enabled
             if (restoreRequest.Project.RestoreMetadata.Sources.Count > 1 && !restoreRequest.PackageSourceMapping.IsEnabled)
             {
-                await _logger.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1507, Strings.Warning_CentralPackageVersions_MultipleSourcesWithoutPackageSourceMapping));
+                await _logger.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1507, string.Format(CultureInfo.CurrentCulture, Strings.Warning_CentralPackageVersions_MultipleSourcesWithoutPackageSourceMapping, restoreRequest.Project.RestoreMetadata.Sources.Count)));
             }
 
             // The dependencies should not have versions explicitly defined if cpvm is enabled.

--- a/test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs
+++ b/test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs
@@ -1680,6 +1680,7 @@ namespace NuGet.Commands.Test
                 else
                 {
                     logger.WarningMessages.Should().Contain(i => i.Contains(NuGetLogCode.NU1507.ToString()));
+                    logger.WarningMessages.Should().Contain(i => i.Contains("There are 3 package sources defined in your configuration"));
                 }
             }
         }

